backend/xivdpscalc/character/__init__.py

```diff
@@ -24,96 +24,140 @@ class Character:
         """
-        Calculates the estimated DPS based on the team composition and current character stats
-        :param potency: Potency calculated on expected rotation
-        :param comp: Team composition. (character.jobs.Comp object)
-        :param is_dot: The potency is applying to a DoT effect (which is additionally modified by spellspeed)
-        :param crit_rate: Overrides crit rate, but not the crit stat.  Calc uses crit stat to calculate rate if this is None
-        :param dh_rate: Overrides direct rate, but not the dh stat.  Calc uses dh stat to calculate rate if this is None
-        :returns: the DPS number
+        Calculates the damage from non probablistic stats.
+        :param potency_per_second: From potency calculated on expected rotation
+        :param comp: Team composition, if applicable
+        :param is_dot: If potency also applies to DoT
+        :returns: DPS contribution from non-probablistic stats
         """
-        # modify mainstat according to number of roles
-        modified_mainstat = Stat(Stats.MAINSTAT, math.floor(self.stats[Stats.MAINSTAT].value * (
-            1 + 0.01 * comp.n_roles)))
-
-        # damage effect of non-probabalistic stats
-        job_mod_scaling = (340 * self.job.job_mod // 1000)
-        damage = potency * (
-            self.wd + job_mod_scaling) * (100 + modified_mainstat.get_multiplier()) // 100
-        damage = self.stats[Stats.DET].apply_stat(damage)
-        damage = self.stats[Stats.TEN].apply_stat(damage)
+
+        # modify mainstat according to number of roles if with a team composition
+        if comp is not None:
+            unique_role_bonus = 1 + 0.01 * comp.n_roles
+            new_mainstat_value = math.floor(self.character_stats[Stats.MAINSTAT].value * unique_role_bonus)
+            mainstat = Stat(Stats.MAINSTAT, new_mainstat_value)
+        else:
+            mainstat = self.character_stats[Stats.MAINSTAT]
+
+        # job bonus for weapon damage
+        adjusted_wd = self.wd + (340 * self.job.job_mod // 1000)
+        np_damage = potency_per_second * adjusted_wd * (100 + mainstat.get_multiplier()) // 100
+
+        # applying the damage bonus from wimpy secondary stats
+        np_damage = self.character_stats[Stats.DET].apply_stat(np_damage)
+        np_damage = self.character_stats[Stats.TEN].apply_stat(np_damage)
         if is_dot:
-            damage = self.stats[Stats.SPEED].apply_stat(damage)
+            np_damage = self.character_stats[Stats.SPEED].apply_stat(np_damage)
 
-        damage //= 100  # why? i do not know. cargo culted
+        return np_damage
 
-        # damage effect of job traits / stance
-        # todo: pull out traits
-        if self.job.role == Roles.HEALER:
-            damage = math.floor(damage * 1.3)  # magic and mend
+    def calc_probablistic_damage(self, np_damage: float, comp: Comp = None, crit_rate: float = None,
```

Comment:
Same thing here: I'd do `Optional[Comp]`

Reply:
Oh, and I missed the `Optional[float]` as well for `crit_rate`

Reply:
fixed